lib/galaxy/managers/base.py

```diff
@@ -1295,10 +1296,8 @@ def raise_filter_err(attr, op, val, msg):
 
 
 def is_valid_slug(slug):
-    """Returns true iff slug is valid."""
-
-    VALID_SLUG_RE = re.compile(r"^[a-z0-9\-]+$")
-    return VALID_SLUG_RE.match(slug)
+    """Returns true if slug is valid."""
+    return slugify(slug, allow_unicode=True) == slug
```

Comment:
Since this is targeting `dev` maybe we can further refactor this.

I feel like this function does not belong in the manager base. A more fitting place would be in the SlugBuilder maybe even as a class method. It feels pretty scattered right now across managers.base, SharableModelManager, SharableMixing, and SlugBuilder.

Just a suggestion feel free to ignore :)

Reply:
Thanks for the suggestion! I agree, it makes sense to move the function to SlugBuilder as a class method.
I am not sure if I did it right: https://github.com/galaxyproject/galaxy/pull/18986/commits/0129b7c5fee3352a45ee87c08a6d89f1281a7485